crates/rust-analyzer/src/command.rs

```diff
@@ -14,23 +14,60 @@ use process_wrap::std::{StdChildWrapper, StdCommandWrap};
 use stdx::process::streaming_output;
 
 /// Cargo output is structured as a one JSON per line. This trait abstracts parsing one line of
-/// cargo output into a Rust data type.
+/// cargo output into a Rust data type where the data is self-describing.
 pub(crate) trait ParseFromLine: Sized + Send + 'static {
```

Comment:
Can we just discard this trait in favor of `CargoParser`? This code is already fairly convoluted and this isn't really helping that

Reply:
I guess its fine for now